sklearn/utils/fixes.py

```diff
@@ -122,11 +157,23 @@ class _FuncWrapper:
 
     def __init__(self, function):
         self.function = function
-        self.config = get_config()
         update_wrapper(self, self.function)
 
+    def with_config(self, config):
+        self.config = config
+        return self
+
     def __call__(self, *args, **kwargs):
-        with config_context(**self.config):
+        config = getattr(self, "config", None)
+        if config is None:
+            warnings.warn(
+                "You are using `sklearn.utils.fixes.delayed` without using "
+                "`sklearn.utils.fixes.Parallel`. A default configuration is used "
+                "instead of propagating the user defined configuration.",
+                UserWarning,
+            )
+            config = get_config()
```

Comment:
Maybe the following will lead to fewer surprises:

```suggestion
            config = {}
```

This means that all the workers get the default configuration instead of some of them getting the current thread config and others the default config.